addons/account/models/partner.py

```diff
@@ -107,6 +106,8 @@ def _check_zip(self):
     def _validate_foreign_vat_country(self):
         for record in self:
             if record.foreign_vat:
+                if not record.country_id:
```

Comment:
We can clean up `if record.country_group_id and record.country_id:`.

Reply:
I don't think so, because you can have country_id without country_group_id now.  (the country_group_id is not strictly needed for the logic)

Reply:
But we will always have a `record.country_id` because we just check it here right ? I think we can simplify some of the code here.